vizro-core/src/vizro/integrations/kedro/_data_manager.py

```diff
@@ -3,25 +3,51 @@
 
 from kedro.framework.session import KedroSession
 from kedro.framework.startup import bootstrap_project
-from kedro.io import DataCatalog
+from kedro.io import CatalogProtocol, KedroDataCatalog
+from kedro.pipeline import Pipeline
 
 from vizro.managers._data_manager import pd_DataFrameCallable
 
 
 def catalog_from_project(
     project_path: Union[str, Path], env: Optional[str] = None, extra_params: Optional[dict[str, Any]] = None
-) -> DataCatalog:
+) -> CatalogProtocol | KedroDataCatalog:
     bootstrap_project(project_path)
     with KedroSession.create(
         project_path=project_path, env=env, save_on_close=False, extra_params=extra_params
     ) as session:
         return session.load_context().catalog
 
 
-def datasets_from_catalog(catalog: DataCatalog) -> dict[str, pd_DataFrameCallable]:
-    datasets = {}
-    for name in catalog.list():
-        dataset = catalog._get_dataset(name, suggest=False)
-        if "pandas" in dataset.__module__:
-            datasets[name] = dataset.load
-    return datasets
+def pipelines_from_project(project_path: Union[str, Path]) -> Pipeline:
+    bootstrap_project(project_path)
+    from kedro.framework.project import pipelines
+
+    return pipelines
+
+
+def datasets_from_catalog(
+    catalog: CatalogProtocol | KedroDataCatalog, *, pipeline: Pipeline = None
```

Comment:
@antonymilne I have updated the input type of catalog as per my comment in the issue as `DataCatalog` does not inherit from `CatalogProtocol`.